notebook/static/notebook/js/notebook.js

```diff
@@ -2844,7 +2844,100 @@ define([
+                    click: function() {
+                        var nb_path = d.find('input').val();
+                        var nb_name = nb_path.split('/').slice(-1).pop();
+                        // If notebook name does not contain extension '.ipynb' add it
+                        var ext = utils.splitext(nb_name)[1];
+                        if (ext === '') {
+                            nb_name = nb_name + '.ipynb';
+                            nb_path = nb_path + '.ipynb';
+                        }
+                        var save_thunk = function() {
+                            var model = {
+                                'type': 'notebook',
+                                'content': that.toJSON(),
+                                'name': nb_name
+                             };
+                            return that.contents.save(nb_path, model)
+                                .then(function(data) {
+                                    d.modal('hide');
+                                    that.notebook_name = data.name;
+                                    that.notebook_path = data.path;
+                                    that.session.rename_notebook(data.path);
+                                    that.events.trigger('notebook_renamed.Notebook', data);
+
+                                },function(error) {
+                                   console.error(i18n.msg._(error.message || 'Unknown error saving notebook'));
+                                });
+                        }
+                        that.contents.get(nb_path, {type: 'notebook'}).then(function(data) {
+                            var warning_body = $('<div/>').append(
+                                $("<p/>").text(i18n.msg._('Notebook with that name exists.')));
+                            dialog.modal({
+                                title: 'Save As',
+                                body: warning_body,
+                                buttons: {Cancel: {},
+                                Overwrite: {
+                                    class: 'btn-warning',
+                                    click: function() {
+                                        return save_thunk();
+                                    }
+                                }
+                            }
+                            });
+                        }, function(err) {
+                            return save_thunk();
+                        })
+                        return false;
+                    }
+                },
+            },
+            open : function () {
+                d.find('input[type="text"]').keydown(function (event) {
+                    if (event.which === keyboard.keycodes.enter) {
+                        d.find('.btn-primary').first().click();
+                        return false;
+                    }
+                });
+                d.find('input[type="text"]').focus().select();
```

Comment:
This currently selects the path pre-filled in the input, which is not ideal. Can we make it put the cursor at the end of the input already there?